ssl/ssl_ciph.c

```diff
@@ -1347,9 +1347,11 @@ static int update_cipher_list(STACK_OF(SSL_CIPHER) **cipher_list,
                               STACK_OF(SSL_CIPHER) **cipher_list_by_id,
                               STACK_OF(SSL_CIPHER) *tls13_ciphersuites)
 {
+    STACK_OF(SSL_CIPHER) *tmp_cipher_list;
     int i;
-    STACK_OF(SSL_CIPHER) *tmp_cipher_list = sk_SSL_CIPHER_dup(*cipher_list);
 
+    tmp_cipher_list = (*cipher_list == NULL) ? sk_SSL_CIPHER_new_null() :
```

Comment:
Does this mean we get no TLSv1.2 ciphersuites by default if you explicitly set some TLSv1.3 ciphersuites? Shouldn't we instead inherit the default settings from the SSL_CTX?

Reply:
Ya, currently `s->tls13_ciphersuites` are inherited from `SSL_CTX` in `SSL_new()`, but not `s->cipher_list`. I thought this is intentionally done.

Reply:
It's done "on demand" via a call to `SSL_get_ciphers`:
https://github.com/openssl/openssl/blob/a6dfa188204f81231a9d25cd007d0753657bcc1a/ssl/ssl_lib.c#L2558-L2568

It would be good to add a test for this scenario, i.e. SSL_set_cipher_list() not called, but SSL_set_ciphersuites() is called...we should get the the default TLSv1.2 ciphers from SSL_CTX, and the explicitly set TLSv1.3 ciphers.

Reply:
With current fix, calling only `SSL_set_ciphersuites` will not inherit TLSv1.2 ciphersuites from `SSL_CTX`. If that is the expectation then I need to change the fix and add test for this.

Reply:
I think that is the expectation.

Reply:
Changes are done. Please review.